CDCgov/NEDSS-Modernization
# CNFT2-1603 configurable page builder features

## Description

The following allows the PageBuilder `Page Library` and `Page Management` features to be locked behind a feature flag. This PR will need to be handled in parallel with [Helm chart PR](https://github.com/CDCgov/NEDSS-Helm/pull/181).

## Tickets

* [CNFT2-1603](https://cdc-nbs.atlassian.net/browse/CNFT2-1603)

Steps to verify:
1. Restart containers
2. Ensure Page Library and Page Management (create page) route to modernized versions
3. Stop containers
4. Launch modernization-api with the following command
```sh
./gradlew :modernization-api:bootRun --args='--server.port=9080 --nbs.ui.features.pageBuilder.enabled=true --nbs.ui.features.pageBuilder.page.library.enabled=true --nbs.ui.features.pageBuilder.page.management.enabled=false'
```
5. Launch nbs-gateway with the following command
```sh
./gradlew :nbs-gateway:bootRun --args='--nbs.gateway.pagebuilder.enabled=true --nbs.gateway.pagebuilder.page.library.enabled=true --nbs.gateway.pagebuilder.page.management.enabled=false'
```
6. Start the UI with `npm run start`
7. Configure [Traefik to use these "local" services (non-docker)](https://github.com/CDCgov/NEDSS-Modernization?tab=readme-ov-file#running-with-local-servers)
```sh
NBS_GATEWAY_SERVER=host.docker.internal MODERNIZATION_UI_SERVER=host.docker.internal MODERNIZATION_UI_PORT=3000 MODERNIZATION_API_SERVER=host.docker.internal MODERNIZATION_API_PORT=9080  docker-compose up reverse-proxy -d
```
8. Verify the modernized Page Library is available
9. Verify clicking on a page name routes to Classic NBS
10. Verify clicking `Create new page` routes to Classic NBS
11. Verify clicking Page Library from Classic routes to Modernized 
12. Change feature flags as desired and verify outcome is as expected


## Checklist before requesting a review


[CNFT2-1603]: https://cdc-nbs.atlassian.net/browse/CNFT2-1603?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
